src/solver/solver.go

```diff
@@ -570,3 +581,18 @@ func getClassOrTrait(typeName string) (meta.ClassInfo, bool) {
 	}
 	return class, false
 }
+
+// replaceTraitName replaces traitName with className inside res.
+//
+// Note: since we don't have a proper class context in non-static method
+// calls as well as other instance-related operations, we can't use this
+// outside of the static context.
```

Comment:
I don't really see why we couldn't do the same in dynamic contexts though. We always have a class name and a trait name when calling a method, right? It should require a more sophisticated approach than the one used here but I think it should still be possible as the end result should be the same: we replace a resulting trait name with the class that used that trait last.

Reply:
I removed that comment part.

Reply:
Let's revisit non-static contexts after the need arises. It should be doable, as you said.